src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -163,7 +163,7 @@ IEnumerator IEnumerable.GetEnumerator()
         public IPartition<TElement> Skip(int count)
         {
             int minIndex = _minIndex + count;
-            return minIndex >= _maxIndex ? EmptyPartition<TElement>.Instance : new OrderedPartition<TElement>(_source, minIndex, _maxIndex);
+            return (uint)minIndex > (uint)_maxIndex ? EmptyPartition<TElement>.Instance : new OrderedPartition<TElement>(_source, minIndex, _maxIndex);
```

Comment:
That might not be the worse idea.

Indeed, while I've been busy with things that have kept me away from contributing and hence had the sort of ideas one only has while not working on stuff, it's occurred to me that some of these (but not this case) could be better with a `_skipped` and `_taken`, if the source is not inherently limited to being less than `int.MaxValue` in size. This isn't an example of that as `OrderBy()` can't handle larger sources anyway, but some of the others might be. I had looking at that as the next thing I wanted to look at after some Expressions tests. Maybe I should bump it, and double-checking all the other cases, up my list :smile: